cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/system/SystemWorkerServiceLauncher.java

```diff
@@ -150,6 +153,27 @@ public void run() {
                     ArtifactLocalizerTwillRunnable.class.getSimpleName());
           }
 
+          if (twillPreparer instanceof SecureTwillPreparer) {
+            SecurityContext securityContext = createSecurityContext();
+            twillPreparer = ((SecureTwillPreparer) twillPreparer)
+                .withSecurityContext(SystemWorkerTwillRunnable.class.getSimpleName(), securityContext);
+            // Mount secret in ArtifactLocalizer sidecar which only run trusted code,
+            // so requests originated by ArtifactLocalizer can run with system identity when internal auth
+            // is enabled.
+            twillPreparer = ((SecureTwillPreparer) twillPreparer)
+                .withSecretDisk(ArtifactLocalizerTwillRunnable.class.getSimpleName(),
+                    new SecretDisk(cConf.get(Constants.Twill.Security.MASTER_SECRET_DISK_NAME),
+                        cConf.get(Constants.Twill.Security.MASTER_SECRET_DISK_PATH)));
+            if (cConf.getBoolean(Constants.Twill.Security.WORKER_MOUNT_SECRET)) {
```

Comment:
Why are you using WORKER_MOUNT_SECRET? This flag is for task workers.

Reply:
Thanks for catching this. This was set for true for both rbac(overriden) and non rbac(by default) cases. But it makes sense to not use it. Removed